setup.py

```diff
@@ -1,15 +1,80 @@
 #!/usr/bin/env python
+import os
+import subprocess
+import sys
+from glob import glob
+from setuptools import setup, find_packages, Extension
 
-try:
-    from setuptools import setup
-except ImportError:
-    from distribute_setup import use_setuptools
-    use_setuptools()
-    from setuptools import setup
+
+if os.path.exists('relic'):
+    sys.path.insert(1, 'relic')
+    import relic.release
+else:
+    try:
+        import relic.release
+    except ImportError:
+        try:
+            subprocess.check_call(['git', 'clone',
+                'https://github.com/jhunkeler/relic.git'])
+            sys.path.insert(1, 'relic')
+            import relic.release
+        except subprocess.CalledProcessError as e:
+            print(e)
+            exit(1)
+
+
+version = relic.release.get_info()
+relic.release.write_template(version, 'lib/reftools')
 
 setup(
-    setup_requires=['d2to1>=0.2.9', 'stsci.distutils>=0.3.2'],
-    d2to1=True,
-    use_2to3=False,
-    zip_safe=False
+    name = 'reftools',
+    version = version.pep386,
```

Comment:
I am still confused. Let's say:

1. Tagged 1.7.
2. Added `foo()` on Jan 1.
3. Added `bar()` on Jan 2.
4. Tag 2.0 on Jan 5.
5. Changed `foo()` to `foo2()` on Feb 1.

If user install from source on Jan 3, `foo()` and `bar()` would be reported as included in v1.7. But if user install on Jan 6, then they would be reported as included in v2 but not v1.7? Does `foo2()` belong to v2 or the next release?